android/app/src/main/java/com/gdgnitsurat/quickshare/utils/FirebaseUtil.kt

```diff
@@ -10,16 +14,31 @@ object FirebaseUtil {
 
     private var firebaseAuth: FirebaseAuth = FirebaseAuth.getInstance()
     private var firebaseUser: FirebaseUser? = firebaseAuth.currentUser
+    private val database: FirebaseDatabase = FirebaseDatabase.getInstance()
+    private var deviceid :String = ""
 
-    fun addCurrentUserToFirebaseDatabase() {
-        val database: FirebaseDatabase = FirebaseDatabase.getInstance()
-        val databaseRef: DatabaseReference = database.reference
-        val user = User(firebaseAuth.currentUser?.displayName, firebaseAuth.currentUser?.email)
-        databaseRef.child("users").child(firebaseAuth.currentUser?.uid).setValue(user)
+    fun addCurrentUserToFirebaseDatabase(deviceID:String) {
+        val databaseRef: DatabaseReference = database.getReference("users")
+        val user = User(firebaseAuth.currentUser?.displayName, firebaseAuth.currentUser?.email, deviceID)
+        databaseRef.child(firebaseAuth.currentUser?.uid).setValue(user)
+    }
+
+    fun addDevicesToFirebaseDatabase(deviceID:String) {
+        deviceid = deviceID
+        val databaseRef: DatabaseReference = database.getReference("devices")
+        val device = Device("clips","android", firebaseAuth.currentUser?.uid)
+        databaseRef.child(deviceID).setValue(device)
+    }
+
+    fun addClipsToFirebaseDatabase(clip:String,time:String) {
```

Comment:
Clips -> Clip

Reply:
Same for this also

Reply:
function is taking only one clip as parameter so it seems little confusing

Reply:
I am going to update...
you'll be clear soon